gluten-flink/ut/src/test/java/org/apache/gluten/streaming/api/operators/GlutenStreamJoinOperatorTestBase.java

```diff
@@ -0,0 +1,338 @@
+import java.util.Queue;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.apache.gluten.streaming.api.operators.utils.RowDataTestUtils.checkEquals;
+
+public abstract class GlutenStreamJoinOperatorTestBase extends StreamingJoinOperatorTestBase {
+
+  protected static RowType leftVeloxType;
+  protected static RowType rightVeloxType;
+  protected static RowType outputVeloxType;
+  protected static org.apache.flink.table.types.logical.RowType outputRowType;
+
+  protected MemoryManager sharedMemoryManager;
+  protected Session sharedSession;
+  protected BufferAllocator sharedAllocator;
+
+  @BeforeAll
+  public static void setupGlutenEnvironment() {
+    Velox4jEnvironment.initializeOnce();
+    setGlutenTypes();
+  }
+
+  private static void setGlutenTypes() {
+
+    InternalTypeInfo<RowData> leftTypeInfo =
+        InternalTypeInfo.of(
+            org.apache.flink.table.types.logical.RowType.of(
+                new LogicalType[] {
+                  new CharType(false, 20), new CharType(false, 20), VarCharType.STRING_TYPE
+                },
+                new String[] {"order_id", "line_order_id", "shipping_address"}));
+    InternalTypeInfo<RowData> rightTypeInfo =
+        InternalTypeInfo.of(
+            org.apache.flink.table.types.logical.RowType.of(
+                new LogicalType[] {new CharType(false, 20), new CharType(true, 10)},
+                new String[] {"line_order_id0", "line_order_ship_mode"}));
+
+    leftVeloxType = (RowType) LogicalTypeConverter.toVLType(leftTypeInfo.toRowType());
+    rightVeloxType = (RowType) LogicalTypeConverter.toVLType(rightTypeInfo.toRowType());
+
+    outputRowType =
+        org.apache.flink.table.types.logical.RowType.of(
+            Stream.concat(
+                    leftTypeInfo.toRowType().getChildren().stream(),
+                    rightTypeInfo.toRowType().getChildren().stream())
+                .toArray(LogicalType[]::new),
+            Stream.concat(
+                    leftTypeInfo.toRowType().getFieldNames().stream(),
+                    rightTypeInfo.toRowType().getFieldNames().stream())
+                .toArray(String[]::new));
+
+    outputVeloxType = (RowType) LogicalTypeConverter.toVLType(outputRowType);
+  }
+
+  @Override
+  @BeforeEach
+  public void beforeEach(TestInfo testInfo) {
+    sharedMemoryManager = MemoryManager.create(AllocationListener.NOOP);
+    sharedSession = Velox4j.newSession(sharedMemoryManager);
+    sharedAllocator = new RootAllocator(Long.MAX_VALUE);
+  }
+
+  @Override
+  @AfterEach
+  public void afterEach() {
+    try {
+      if (sharedAllocator != null) {
+        sharedAllocator.close();
+        sharedAllocator = null;
+      }
+      if (sharedSession != null) {
+        sharedSession.close();
+        sharedSession = null;
+      }
+      if (sharedMemoryManager != null) {
+        sharedMemoryManager.close();
+        sharedMemoryManager = null;
+      }
+    } catch (Exception e) {
```

Comment:
Seems unnecessary to add the catch. Maybe, just let the test fail if exception happens. If there is checked exception, just declare the method with throws.